### PR TITLE
:new: 2 Adapters & 2 Chains

### PR DESCRIPTION
**Pull request by ftm1337**
Continued from https://github.com/DefiLlama/DefiLlama-Adapters/pull/2886
### Changelog
 - [x] Merge all Guru Network adapters into a single multichain adapter with more new chains
 - [x] New adapter for Value Added eXchange (VAX)
### Todo
 - [ ] Delete adapters in `kccguru` & `ftmguru`
*haven't deleted them to avoid deleting historical data. Requesting further guidance about history preservation..*

----

### 2. Guru
*We pull TVL from a smart contract on-chain from a `function tvl() public view returns(uint256)` which gives USD value of assets locked on our platform. As TVL comes from multiple contracts, and we keep adding more and more contracts as time goes by. we dont want to bug data aggregators with a new request every week. Thats why we had to take this route, which preserves on-chain authenticity and real-time updates. The `tvlGuru` "Universal TVL Finder" loops over each protocol's contracts ("Bank") which holds "Asset"(s). Then it gets the price for asset from the most liquid AMM DEX. Then it checks bank's balance and multiplies with the price. The price calculation methodology varies by asset, as some have direct eth pairs, some have stablecoin pairs and some are token-token pairs, while some are LP tokens, for which we use a mix of above routes to determine real-time the prices at each block.*

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/Guru exports TVL: 
        
 
 
    fantom-pool2              0
    echelon                   10.02 k
    echelon-pool2             0
    fantom                    45.55 k
    echelon-staking           0
    kcc-staking               0
    fantom-staking            0
    kcc                       2.96 k
    kcc-pool2                 0
    multivac                  464.8621337165878
    multivac-pool2            0
    multivac-staking          0
    pool2                     0
    staking                   0
    
    total                    58.99 k

**Comment by llamatester** [comment]
The adapter at projects/VAX exports TVL: 
        
 
 
    multivac                  3.78 k
    
    total                    3.78 k

File: projects/Guru/index.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+//TODO: Create mappings for bridged tokens on new chains
+const USD = {
+	"ftm"	:	{"addr":"fantom:0x04068DA6C83AFCFA0e13ba15A6696662335D5B75",	"deci":6},
+	"kcc"	:	{"addr":"kcc:0x0039f574eE5cC39bdD162E9A88e3EB1f111bAF48",	"deci":18},
+	"mtv"	:	{"addr":"0x6B175474E89094C44Da98b954EedeAC495271d0F",		"deci":18},	//Rep. Mainnet DAI
+	"ech"	:	{"addr":"0x6B175474E89094C44Da98b954EedeAC495271d0F",		"deci":18}	//Rep. Mainnet DAI
+}
+
+
+async function ftm_p_2(timestamp,block) { return {[USD.ftm.addr]:(Number( ( await sdk.api.abi.call({ target: tvlGuru.ftm, abi: ITVL[0], block: block, chain: 'fantom' }) ).output )/(10**(18-USD.ftm.deci)))} }
```
Comment (waynebruce0x) [inline projects/Guru/index.js:81]: you should use chainBlocks.fantom etc here because you're pointing at the eth current block

**Comment by llamatester** [comment]
The adapter at projects/Guru exports TVL: 
        
 
 
    fantom-staking            0
    fantom-pool2              0
    echelon                   8.18 k
    echelon-pool2             0
    fantom                    33.83 k
    kcc-staking               0
    echelon-staking           0
    multivac-pool2            0
    kcc                       2.44 k
    kcc-pool2                 0
    multivac-staking          0
    multivac                  2.90 k
    staking                   0
    pool2                     0
    
    total                    47.35 k

**Comment by llamatester** [comment]
The adapter at projects/VAX exports TVL: 
        
 
 
    multivac                  5.56 k
    
    total                    5.56 k

**Comment by waynebruce0x** [comment]
Do those numbers look right? chainBlocks should be the third param passed into the tvl function

**Comment by llamatester** [comment]
The adapter at projects/Guru exports TVL: 
        
 
 
    fantom-staking            0
    fantom-pool2              0
    fantom                    29.44 k
    multivac-staking          0
    kcc                       2.31 k
    multivac-pool2            0
    echelon-pool2             0
    echelon-staking           0
    kcc-staking               0
    multivac                  2.94 k
    echelon                   8.06 k
    kcc-pool2                 0
    staking                   0
    pool2                     0
    
    total                    42.74 k

**Comment by llamatester** [comment]
The adapter at projects/VAX exports TVL: 
        
 
 
    multivac                  5.07 k
    
    total                    5.07 k

**Comment by ftm1337** [comment]
Updated the adapter with the proper parameters.
yes @waynebruce0x, the numbers look right. we lost significant tvl recently which made this look different from the numbers in the initial comment from :llama::robot:. we're working towards regaining it :relieved:

**Comment by llamatester** [comment]
The adapter at projects/Guru exports TVL: 
        
 
 
    fantom-pool2              0
    fantom                    32.55 k
    fantom-staking            0
    echelon-pool2             0
    kcc                       2.45 k
    kcc-staking               0
    multivac-staking          0
    echelon-staking           0
    kcc-pool2                 0
    echelon                   8.77 k
    multivac                  2.98 k
    multivac-pool2            0
    pool2                     0
    staking                   0
    
    total                    46.76 k

**Comment by llamatester** [comment]
The adapter at projects/VAX exports TVL: 
        
 
 
    multivac                  5.65 k
    
    total                    5.65 k

**Pull request merged**